app/uk/gov/hmrc/cardpaymentfrontend/models/Address.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cardpaymentfrontend.models
+
+import play.api.libs.json.{Json, OFormat}
+
+final case class Address(
+    line1:    String,
+    line2:    Option[String] = None,
+    city:     Option[String] = None,
+    county:   Option[String] = None,
+    postcode: String,
+    country:  String
+) {
+  def hasSelect(maybeString: Option[String]): Boolean = {
+    maybeString match {
+      case Some(s) =>
+        val selectPattern = "(?i)select".r
+        selectPattern.findFirstIn(s).isDefined
+      case None => false
+    }
+  }
+
+  // For UK (GBR) addresses only, replace counties containing variations on Select with None.
```

Comment:
From recollection, wasn't this a problem for Barclaycard in International addresses?